pyvista/core/datasetattributes.py

```diff
@@ -611,6 +618,13 @@ def set_array(
 
         """
         vtk_arr = self._prepare_array(data, name, deep_copy)
+
+        # remove complex association if dataset is not complex
+        complex_assoc = self.dataset._association_complex_names
+        if name in complex_assoc[self.association.name]:
+            if not isinstance(data, np.ndarray) or not np.issubdtype(data.dtype, complex):
```

Comment:
I know this is an unlikely edge case, but will anyone pass a list of complex values? If we want to anticipate that might be a bit safer to call `np.asarray()` on the data and check its dtype. I think `_prepare_array()` also converts iterables (should probably be sequences!) to arrays before checking complexness, so this would be safer.

And again I'd check `issubdtype(data.dtype, np.complexfloating)`, should we ever decide to allow quad precision complexes.

Reply:
I've removed this and placed the check within `_prepare_array` since we convert to a `numpy.ndarray` there already.